include/mbedtls/ecdh.h

```diff
@@ -127,6 +181,24 @@ int mbedtls_ecdh_compute_shared( mbedtls_ecp_group *grp, mbedtls_mpi *z,
  */
 void mbedtls_ecdh_init( mbedtls_ecdh_context *ctx );
 
+/**
+ * \brief           This function sets up the ECDH context with the information
+ *                  given.
+ *
+ *                  This function should be called after mbedtls_ecdh_init() but
+ *                  before mbedtls_ecdh_make_params(). There is no need to call
+ *                  this function before mbedtls_ecdh_read_params().
+ *
+ *                  This is the first function used by a TLS server for ECDHE
+ *                  ciphersuites.
+ *
+ * \param ctx       The ECDH context to set up.
+ * \param grp       The group id of the group to set up the context for.
+ *
+ * \return          \c 0 on success.
+ */
+int mbedtls_ecdh_setup( mbedtls_ecdh_context *ctx, mbedtls_ecp_group_id grp );
```

Comment:
Old applications aren't going to call `mbedtls_ecdh_setup()`. How will ECDH still work for old applications that aren't changing their code? Is it unnecessary to call `mbedtls_ecdh_setup()` when using the legacy context configuration? If it is now necessary, this is an API break in the default configuration, which we potentially care to avoid.

Reply:
Old applications had to do by hand all the work that `mbedtls_ecdh_setup()` does. In the legacy context configuration it works either way. Old applications do not need to change their code, new applications may use `mbedtls_ecdh_setup()` or set it by hand like the old ones.

Reply:
OK